MovieBooking/Modules/Authentication/ViewModel/AuthenticationViewModel.swift

```diff
@@ -0,0 +1,74 @@
+import UIKit
+import FirebaseAuth
+import FirebaseFirestore
+
+class AuthenticationViewModel {
+    
+    // MARK: - Variables
+    var logInValidationError: ((String) -> Void)?
+    var signUpValidationError: ((String) -> Void)?
+    var onLogInSuccess: ((String) -> Void)?
+    var onSignUpSuccess: ((String) -> Void)?
+    
+    // MARK: - File Private Functions
+    fileprivate func validateFieldsforLogIn(_ email: String, _ password: String) -> String? {
+        if email.isEmpty {
+            return "Email is Empty"
+        } else if password.isEmpty {
+            return "Password is Empty"
+        } else if password.count < 8 {
+            return "Need atleast 8 characters"
+        } else if !email.contains("@") || !email.contains(".") {
+            return "Wrong email"
+        }
+        return nil
+    }
+    
+    func doLogIn(_ email: String, _ password: String) {
+        let validationResult = validateFieldsforLogIn(email, password)
+        if let validationResult = validationResult {
+            logInValidationError?(validationResult)
+        } else {
+            Auth.auth().signIn(withEmail: email, password: password) { [weak self] authResult, error in
+                guard self != nil else { return }
```

Comment:
you can write it like guard let `self` = self else { return } or else  guard let uSelf = self else { return }